### PR TITLE
CMake updates

### PR DESCRIPTION
**Pull request by mplooster**
Modified CMake (c++ sdk) so that it would have better support for Windows. Also added a brief section in the C++ README about building on Windows.

### CODE REVIEW

File: grpc/cpp-cubic/CobaltGRPC.cmake [hunk 1]
```diff
@@ -5,139 +5,128 @@
...
+        DEPENDS ${ALL_PROTOS})
+
+    # Make sure the output variable specified makes it back to the
+    # parent scope.
+    set(${RUN_PROTOC_RESULT_FILES} ${${RUN_PROTOC_RESULT_FILES}} PARENT_SCOPE)
+
+endfunction(run_protoc)
 
-# Check if we have already defined the run_protoc function
-if(NOT COMMAND run_protoc)
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/CobaltGRPC.cmake:57]: Just for my education, why do we no longer want to check whether `run_protoc` is already defined?

File: grpc/cpp-cubic/README.md [hunk 2]
```diff
@@ -58,6 +58,20 @@ FetchContent_Populate(sdk_cubic)
 add_subdirectory(${sdk_cubic_SOURCE_DIR}/grpc/cpp-cubic ${sdk_cubic_BINARY_DIR})
 ```
 
+### Windows Build
+When building for Windows, we recommend running CMake in the Visual
+Studio [command line](https://docs.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2019).
+The VS command line will automatically set several environment variables
+that will be useful to CMake in defining compilers and other options.
+You will also need to download and install the Netwide Assmbler from
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/README.md:66]:
```suggestion
You will also need to download and install the Netwide Assembler from
```

File: grpc/cpp-cubic/README.md [hunk 3]
```diff
@@ -58,6 +58,20 @@ FetchContent_Populate(sdk_cubic)
...
+[here](https://www.nasm.us/). Make sure it is available to your `PATH`
+so that CMake can find it (or specify it manually in the CMake
+configuration).
+
+Also note that gRPC doesn't really support building shared libraries
+on Windows. If you try it, you will likely get build errors, similar to
+what is reported [here](https://github.com/grpc/grpc/issues/25311).
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/README.md:73]: So does that mean that the default cmake build on Windows will build a static library?  If so, we should say so to reassure them that they'll be OK as long as they don't try to do anything fancy.  Or if there are particular config options they need to pass or avoid passing, specify them.

**Review by renaissanceGeek — APPROVED** [review]
I don't pretend to be an expert, but as far as I can tell, it makes sense :-)

File: grpc/cpp-cubic/CobaltGRPC.cmake (hunk 1 above)
Comment (mplooster) [reply grpc/cpp-cubic/CobaltGRPC.cmake:57]: I think I decided at some point that I didn't care if it was defined or not, I wanted to use my `run_protoc` command. (I may have also decided it was very unlikely for the function to already be defined).

File: grpc/cpp-cubic/README.md (hunk 3 above)
Comment (mplooster) [reply grpc/cpp-cubic/README.md:73]: Yes. The default will build a static library. It probably wouldn't hurt to mention that they shouldn't add `-DBUILD_SHARED_LIBS=TRUE` to the cmake command. Or if they do, to make it explicitly off `-DBUILD_SHARED_LIBS=FALSE`.

**Review by happyalu — APPROVED** [review]
lgtm.

**Pull request merged**